File: eio2024-ev/2024-12-08-ev/sokid/solution/sol3.cpp
#include <iostream>
#include <string>
#include <set>
using namespace std;

int main() {
	int n;
	cin >> n;

	set<string> yksikud;
	for (int i = 1; i <= n; ++i) {
		string varv;
		cin >> varv;
		if (yksikud.contains(varv)) {
			yksikud.erase(varv);
		} else {
			yksikud.insert(varv);
		}
	}

	if (n % 2 > 0){
		cout << -1 << '\n';
	} else{
		cout << yksikud.size() / 2 << '\n';
	}
}

File: eio2024-ev/2024-12-08-ev/sokid/solution/sol3.py
n = int(input())

yksikud = set()
for i in range(n):
    varv = input().strip()
    if varv in yksikud:
        yksikud.remove(varv)
    else:
        yksikud.add(varv)

if n % 2 > 0:
    print(-1)
else: 
    print(len(yksikud) // 2)
